### ISSUE TITLE
PSCommand.Clone strips `CommandInfo` from inner `Command` objects

### ISSUE THREAD

**Issue by SeeminglyScience** [PowerShell/PowerShell#12297]
# Steps to reproduce

```powershell
$ps = $ps2 = $null
try {
    $ps = [powershell]::Create('CurrentRunspace')
    $cmdlet = [System.Management.Automation.CmdletInfo]::new('un-resolvable', [Microsoft.PowerShell.Commands.FormatDefaultCommand])
    $null = $ps.AddCommand($cmdlet).AddParameter("InputObject", 'test')

    $ps2 = [powershell]::Create('CurrentRunspace')

    # Setter for "Commands" calls PSCommand.Clone()
    $ps2.Commands = $ps.Commands

    $ps.Invoke()
    $ps2.Invoke()
} finally {
    if ($null -ne $ps2) {
        $ps2.Dispose()
    }

    if ($null -ne $ps) {
        $ps.Dispose()
    }
}
```

# Expected behavior

```none
test
test
```

# Actual behavior

```none
test
MethodInvocationException:
Line |
  13 |      $ps2.Invoke()
     |      ~~~~~~~~~~~~~
     | Exception calling "Invoke" with "0" argument(s): "The term 'un-resolvable' is not recognized as the name of a cmdlet, function, script file, or operable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again."
```

# Environment data

```none
Name                           Value
----                           -----
PSVersion                      7.1.0-preview.1
PSEdition                      Core
GitCommitId                    7.1.0-preview.1
OS                             Microsoft Windows 10.0.18362
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

# Fix

The clone constructor for `SMAR.Command` doesn't copy the value of the `CommandInfo` property from the command to be cloned.

/cc @TylerLeonhardt

**Comment by TylerLeonhardt** [comment]
This bug is called out in my PR here:
https://github.com/PowerShell/PowerShellEditorServices/pull/1258/files#diff-39fdedf9e00646e4c04b7deaa1225bb6

**Comment by iSazonov** [comment]
@TylerLeonhardt Do you want to get the fix in 7.0 Servicing?

**Comment by TylerLeonhardt** [comment]
@iSazonov I've opened a PR: https://github.com/PowerShell/PowerShell/pull/12301
